### PR TITLE
Infinite scrolling

### PR DESCRIPTION
**Pull request by vatrey**
1. Added Infinite Scrolling.
2. Added Loader and End Message with Sad Emoji

Story Link [here](https://abhirams.atlassian.net/jira/software/projects/P3ECWFC/boards/6?assignee=6253d9524f1d57006a23b060&selectedIssue=P3ECWFC-46)

Task Link for infinite Scrolling [here](https://abhirams.atlassian.net/jira/software/projects/P3ECWFC/boards/6?assignee=6253d9524f1d57006a23b060&selectedIssue=P3ECWFC-91)

Task Link for Loader and spinner [here](https://abhirams.atlassian.net/jira/software/projects/P3ECWFC/boards/6?assignee=6253d9524f1d57006a23b060&selectedIssue=P3ECWFC-93)

### CODE REVIEW

File: src/components/Searchbar/Searchbar.tsx [hunk 1]
```diff
@@ -23,7 +24,7 @@ function Searchbar() {
             onChange={handleTextChange}
           />
           <Link to={`/cars?search-text=${searchWord}`}>
-            <button
+            <button disabled={searchWord.length === 0}
```
Comment (pashedin) [inline src/components/Searchbar/Searchbar.tsx]: Haha, I see what you are doing here

File: src/pages/CarsList/CarsList.tsx [hunk 2]
```diff
@@ -55,13 +65,21 @@ function CarsList() {
     window.scrollTo(0, 0);
   }, [location]);
 
+  let fetchMoreData = () => {
+    setTimeout(() => {
+      if (carType == 3) dispatch(getMoreCars(""));
+      else dispatch(getMoreCars(carTypeList[carType]));
+    }, 2000);
```
Comment (pashedin) [inline src/pages/CarsList/CarsList.tsx:82]: Non Actionable: I do not like this idea of you using a set timeout, but ok for now. I hope this is odne only to simulate loading/loaders.

File: src/pages/CarsList/CarsList.tsx [hunk 3]
```diff
@@ -123,7 +144,25 @@ function CarsList() {
...
+            next={fetchMoreData}
+            hasMore={cars?.length < 101}
+            loader={<div className="loader">
+              <Spinner animation="grow" />
+              <span>Loading...</span>
+            </div>}
+            endMessage={<div className="end-msg">
+              <img src={sademoji} alt="Sad Smiley Face" width={200} />
+              <p>There are no more cars</p>
+            </div>}
```
Comment (pashedin) [inline src/pages/CarsList/CarsList.tsx]: Please create a different component for this

File: src/pages/CarsList/CarsList.tsx [hunk 4]
```diff
@@ -123,7 +144,25 @@ function CarsList() {
...
         {carType !== 4 &&
-          cars?.map((car: CarDetails) => <CarCard key={car.id} car={car} />)}
+          <InfiniteScroll
+            dataLength={(cars === undefined) ? 1 : cars?.length}
+            next={fetchMoreData}
+            hasMore={cars?.length < 101}
+            loader={<div className="loader">
+              <Spinner animation="grow" />
+              <span>Loading...</span>
+            </div>}
```
Comment (pashedin) [inline src/pages/CarsList/CarsList.tsx]: Please create a different component for this

**Review by pashedin — CHANGES_REQUESTED** [review]

**Review by pashedin — DISMISSED** [review]

**Comment by pashedin** [comment]
Please resolve conflicts

**Comment by vatrey** [comment]
> Please resolve conflicts

@pashedin Resolved the conflicts

**Review by pashedin — APPROVED** [review]
good job 👍

**Pull request merged**